### PR TITLE
chore: add a bunch of sentry stuff

### PR DESCRIPTION
**Pull request by ajbt200128**
We should now send a bunch of useful info to sentry when we crash, including the crash output from Semgrep! We will also include what settings were set.
PR checklist:

- [X] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [X] Tests included or PR comment includes a reproducible test plan
- [X] Documentation is up-to-date
- [X] A changelog entry was for any user-facing change
- [X] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by tpetr** [comment]
Do we need to update any docs saying that we're collecting error / crash reports now?

**Comment by ajbt200128** [comment]
@tpetr in https://semgrep.dev/docs/metrics we say we  "High-level error and warning classes that Semgrep encounters when run" which I feel is enough? maybe we want to update this page altogether though to mention IDE stuff

File: src/telemetry/sentry.ts [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    }
+  }
+}
+
+// ProxyOutputChannel is just a proxy to a normal output channel
+// This is needed as outputChannels have no way of grabbing their full log
+// There's a Github issue that confirms this is impossible that I can no longer find
+export class ProxyOutputChannel implements vscode.OutputChannel {
+  readonly name: string;
+  private fullLog = "";
```
Comment (tpetr) [inline src/telemetry/sentry.ts:169]: is this the entire semgrep lsp log being stored in memory? how big are these logs going to be? is writing to a temp file and then uploading that file not an option?

**Review by tpetr — APPROVED** [review]
LGTM aside from question about log storage

**Comment by tpetr** [comment]
Can you update the changelog in this PR as well?

**Pull request merged**